src/parallel_parameter_search/walk_optimization.py

```diff
@@ -0,0 +1,508 @@
+    def __init__(self, namespace, gui, walk_as_node, sim_type='pybullet'):
+        super(WolfgangWalkOptimization, self).__init__(namespace, 'wolfgang', walk_as_node)
+        self.reset_height_offset = 0.005
+        self.directions = [[0.1, 0, 0],
+                           [-0.1, 0, 0],
+                           [0, 0.05, 0],
+                           [0, -0.05, 0],
+                           [0, 0, 0.5],
+                           [0, 0, -0.5],
+                           [0.1, 0, 0.5],
+                           [0, 0.05, -0.5],
+                           [0.1, 0.05, 0.5],
+                           [-0.1, -0.05, -0.5]
+                           ]
+        if sim_type == 'pybullet':
+            self.sim = PybulletSim(self.namespace, gui)
+        elif sim_type == 'webots':
+            # todo
+            self.sim = WebotsSim(self.namespace, gui)
+        else:
+            print(f'sim type {sim_type} not known')
+
+    def suggest_walk_params(self, trial):
+        param_dict = {}
+
+        def add(name, min_value, max_value):
+            param_dict[name] = trial.suggest_uniform(name, min_value, max_value)
+
+        add('double_support_ratio', 0.0, 0.45)
+        add('freq', 1.5, 3)
+        add('foot_distance', 0.1, 0.3)
+        add('trunk_height', 0.38, 0.45)
+        add('trunk_phase', -0.5, 0.5)
+        add('trunk_swing', 0.0, 1.0)
+        add('trunk_x_offset', -0.03, 0.03)
+
+        add('trunk_x_offset_p_coef_forward', -1, 1)
+        add('trunk_x_offset_p_coef_turn', -1, 1)
+
+        # add('first_step_swing_factor', 0.0, 2)
+        # add('first_step_trunk_phase', -0.5, 0.5)
+        param_dict['first_step_swing_factor'] = 1
+        param_dict['first_step_trunk_phase'] = -0.5
+
+        # add('foot_overshoot_phase', 0.0, 1.0)
+        # add('foot_overshoot_ratio', 0.0, 1.0)
+        param_dict['foot_overshoot_phase'] = 1
+        param_dict['foot_overshoot_ratio'] = 0.0
+
+        # add('trunk_y_offset', -0.03, 0.03)
+        # add('foot_rise', 0.04, 0.08)
+        # add('foot_apex_phase', 0.0, 1.0)
+        param_dict['trunk_y_offset'] = 0
+        param_dict['foot_rise'] = 0.1
+        param_dict['foot_apex_phase'] = 0.5
+        # todo put this as addition arguments to trial
+
+        # add('trunk_pitch', -1.0, 1.0)
+        # add('trunk_pitch_p_coef_forward', -5, 5)
+        # add('trunk_pitch_p_coef_turn', -5, 5)
+        param_dict['trunk_pitch'] = 0
+        param_dict['trunk_pitch_p_coef_forward'] = 0
+        param_dict['trunk_pitch_p_coef_turn'] = 0
+
+        # add('foot_z_pause', 0, 1)
+        # add('foot_put_down_phase', 0, 1)
+        # add('trunk_pause', 0, 1)
+        param_dict['foot_z_pause'] = 0
+        param_dict['foot_put_down_phase'] = 1
+        param_dict['trunk_pause'] = 0
+
+        # todo 'trunk' nochmal ander nennen? body?
+
+        # todo also find PID values, maybe in a second step after finding walking params
+        # todo de/activate phase reset while searching params? yes
```

Comment:
TODOs → Issues
Especially the one with `yes`

Reply:
yeah was some old comments, has been done already